### DISCUSSION TITLE
V7: Custom sorting, sortType, parameter "desc" is undefined.

### DISCUSSION THREAD

**Discussion by MatthieuStadelmann** [TanStack/table#1974] [General]
Hi!
First, thank you for this great library.

I'm currently implementing a custom sorting function that would allow me to keep all the null values at the bottom of the table.

Therefore I created a custom function that looks like this:

` const customSortType = useCallback((rowA, rowB, columnId, desc) => {
        console.log("inside customSortType", desc);
        let a = getRowValueByColumnID(rowA, columnId);
        let b = getRowValueByColumnID(rowB, columnId);
        let dateA = new Date(a.date).getTime();
        let dateB = new Date(b.date).getTime();

        return compareBasic(dateA, dateB)

    }, []);`

Unfortunately, the last parameter is always undefined. Should I initiate it somewhere?

This is how I initiate my table:

`const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    rows,
    prepareRow,
} = useTable({
    columns,
    data,
    initialState: {
        sortBy: [{id: "name", desc: false}]
    }
}, useSortBy);`

And here is the code of my column:

`const COLUMN_LAST_CHECK_UP = {
Header: makeHeader(HEADER_LAST_CHECK_UP),
accessor: "lastCheckUp",
Cell: LastCheckUpCell,
sortType: customSortType
};`

Thanks for your help!

**Comment by gargroh** [comment]
Using latest version ?

**Comment by MatthieuStadelmann** [comment]
"react-table": "^7.0.0-beta.20",

**Comment by gargroh** [comment]
This flag was added near to that version only, not sure that it is included in mentioned version, please use the latest version to resolve your issue.

**Comment by MatthieuStadelmann** [comment]
Thank you @gargroh, it works.

**Comment by DuneArchitect** [comment]
It's unclear from the docs what the sortType function passed in should return.

Looking at [sortTypes.js](https://github.com/tannerlinsley/react-table/blob/127a7fca8754f6289b6bc019dbd1fc4bcdd13187/src/sortTypes.js) it should return 0, -1, or 1. 
```
function compareBasic(a, b) {
  return a === b ? 0 : a > b ? 1 : -1
}
```
Should this simple function perhaps be exposed so that it can be imported? @tannerlinsley
